src/operator/pad-inl.h

```diff
@@ -0,0 +1,192 @@
+/*!
+ * Copyright (c) 2016 by Contributors
+ * \file pad-inl.h
+ * \brief
+ * \author Sebastian Bodenstien
+*/
+
+#ifndef MXNET_OPERATOR_PAD_INL_H_
+#define MXNET_OPERATOR_PAD_INL_H_
+
+#include <dmlc/logging.h>
+#include <dmlc/parameter.h>
+#include <mxnet/operator.h>
+#include <algorithm>
+#include <map>
+#include <vector>
+#include <string>
+#include <utility>
+#include "./operator_common.h"
+#include "./mshadow_op.h"
+
+namespace mxnet {
+namespace op {
+
+namespace pad_enum {
+enum PadOpInputs { kData };
+enum PadOpType { kConstant, kEdge };
+enum PadOpOutputs { kOut };
+}
+
+struct PadParam : public dmlc::Parameter<PadParam> {
+  int mode;
+  double constant_value;
+  TShape pad_width;
+  DMLC_DECLARE_PARAMETER(PadParam) {
+    DMLC_DECLARE_FIELD(mode)
+        .add_enum("constant", pad_enum::kConstant)
+        .add_enum("edge", pad_enum::kEdge)
+        .describe(
+            "Padding type to use. \"constant\" pads all values with a constant "
+            "value,"
+            "the value of which can be specified with the constant_value"
+            "option. \"edge\" uses the boundary values of the array as "
+            "padding.");
+
+    DMLC_DECLARE_FIELD(pad_width).describe(
+        "A tuple of padding sizes of length 2*r, where r is the rank of the "
```

Comment:
Ok, I've improved the docs, and mentioned the relationship between MXNet pad_width and Numpy pad_width.